Shopify/checkout-sheet-kit-swift
# Add support for Web Pixel events

### What are you trying to accomplish?

- Add support for Web Pixel events. The events should be emitted only  after checkout has been presented.
- Emit presented message every time checkout loads (this applies to offsite payments mainly) to keep receiving Web Pixel events.

### How to test

Use the following Storefront config:

```
STOREFRONT_DOMAIN = shop1.shopify.payment-apps-wgvz.jose-alvarez.eu.spin.dev
STOREFRONT_ACCESS_TOKEN = b877504bef9f8b9adb731cea4b1732d2
```

1. Start a checkout using the MobileBuyIntegration sample app
2. Check in the console that Web Pixel events were emitted
3. Pay using "Bogus payments" and complete the offsite payment
4. In the thank you page, check in the console that Web Pixel events are still emitted.


---

### Before you merge

#### Testing

- [x] I've added tests to support my implementation

#### Contribution guidelines

- [x] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CONTRIBUTING.md).
- [x] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CODE_OF_CONDUCT.md).

#### Documentation

- [ ] I've updated any documentation related to these changes.
- [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-swift).

#### Versioning

- [ ] I have added a [Changelog](./CHANGELOG.md) entry.

### After merging

If your changes required a version bump, please add an entry under the [Releases](https://github.com/Shopify/checkout-kit-swift/releases) page - doing so will trigger a CI workflow to publish the latest Cocoapod.
